database/error.go

```diff
@@ -42,3 +42,8 @@ func ErrSQLMapMarshalValue(err error) error {
 func ErrSQLMapUnmarshalScanned(err error) error {
 	return errors.New(ErrSQLMapUnmarshalScannedCode, errors.Alert, []string{"failed to unmarshal scanned data", err.Error()}, []string{err.Error()}, []string{}, []string{})
 }
+
+//ErrClosingDatabaseConnection represents the error which will occur when the database connection fails to get closed
+func ErrClosingDatabaseConnection(err error) error {
+	return errors.New(ErrClosingDatabaseConnectionCode, errors.Alert, []string{"failed to close database connection"}, []string{err.Error()}, []string{"Invalid database instance passed."}, []string{"Make sure the DB handler has a valid database instance."})
```

Comment:
Is this guaranteed to complete when a kill signal is received? If not, the message might need to consider for this.

Also, if not, we need to consider for that in our database connections.

Reply:
In the dev environment, yes it will definitely complete after ctrl+C.  I think this will return easily everytime the os.Interrupt is called.

Reply:
Sounds good.

Reply:
Although if the shell is closed directly, which is the parent process of the running go process, that cannot be handled.

Reply:
"The shell" being a contributor's shell session or being the Docker container's base process?

Reply:
Yes.